src/main/java/org/junitpioneer/jupiter/NotWorkingExtension.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.jupiter.api.extension.LifecycleMethodExecutionExceptionHandler;
+import org.junit.jupiter.api.extension.TestExecutionExceptionHandler;
+import org.opentest4j.TestAbortedException;
+
+class NotWorkingExtension implements Extension, TestExecutionExceptionHandler, LifecycleMethodExecutionExceptionHandler,
+		BeforeEachCallback, AfterEachCallback {
+
+	private static final Namespace NAMESPACE = Namespace.create(NotWorkingExtension.class);
+
+	/**
+	 * {@code Boolean} key indicating whether a 'not working' exception occurred:
+	 *
+	 * <ul>
+	 *   <li>{@code true}: A 'not working' exception occurred; the extension should cause
+	 *   the test to be skipped
+	 *   <li>{@code false}: No exception occurred; the extension should cause the test to fail
+	 *   <li>{@code null}: A regular test abort exception was thrown by user code; the
+	 *   extension should do nothing
+	 * </ul>
+	 */
+	private static final String EXCEPTION_OCCURRED_STORE_KEY = "exceptionOccurred";
+
+	public NotWorkingExtension() {
+	}
+
+	private Store getStore(ExtensionContext context) {
+		return context.getStore(NAMESPACE);
+	}
+
+	private static NotWorking getNotWorkingAnnotation(ExtensionContext context) {
+		return context.getRequiredTestMethod().getDeclaredAnnotation(NotWorking.class);
```

Comment:
If you use Jupiter's `AnnotationSupport::findAnnotation`, it will also look for meta annotations. It returns an `Optional`, which we usually get out of with `.orElseThrow(() -> new IllegalStateException("@NotWorking is missing."))`.